### PR TITLE
VR-4736 Support retrieving a Commit's log

### PR DESCRIPTION
**Pull request by nhatsmrt**

### CODE REVIEW

File: client/scala/src/main/scala/ai/verta/repository/Commit.scala [hunk 1]
```diff
@@ -282,4 +285,20 @@ class Commit(
     else
       Success(newCommit)
   }
+
+  /** Return ancestors, starting from this Commit until the root of the Repository
+   *  @return a list of ancestors
+   */
+  def log()(implicit ec: ExecutionContext): Try[List[Commit]] = {
```
Comment (conradoverta) [inline client/scala/src/main/scala/ai/verta/repository/Commit.scala]: I recommend using a Stream to allow for pagination later.

Comment (nhatsmrt) [reply client/scala/src/main/scala/ai/verta/repository/Commit.scala]: Fixed.

File: client/scala/src/main/scala/ai/verta/repository/Commit.scala [hunk 2]
```diff
@@ -282,4 +285,23 @@ class Commit(
...
+
+    clientSet.versioningService.ListCommitsLog4(
+      repository_id_repo_id = repo.id,
+      commit_sha = commitSHA
+    ) // Try[VersioningListCommitsLogRequestResponse]
+    .map(_.commits) // Try[Option[List[VersioningCommit]]]
+    .map(ls =>
+      if (ls.isEmpty) Stream()
+      else ls.get.toStream.map(c => new Commit(clientSet, repo, c))
+    ) // Try[List[Commit]]
```
Comment (conradoverta) [inline client/scala/src/main/scala/ai/verta/repository/Commit.scala]: Fix comment =)

Comment (nhatsmrt) [reply client/scala/src/main/scala/ai/verta/repository/Commit.scala]: Woops =) Fixed.

**Review by conradoverta — APPROVED** [review]

**Pull request merged**